services/failure_normalizer.py

```diff
@@ -0,0 +1,77 @@
+from typing import Dict
+
+import regex
+
+predefined_dict_of_regexes_to_match = {
+    "UUID": r"[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}",
+    "DATETIME": r"(?:19|20)[0-9]{2}-(?:(?:0?[0-9])|(?:1[012]))-(?:(?:0?[0-9])|1[0-9]|2[0-9]|3[01])T(?:0[0-9]|1[0-9]|2[0-3]):(?:[0-5][0-9]):(?:[0-5][0-9])(?:Z|(?:-(?:0[0-9]:[03]0)))?",
+    "DATETIME2": r"(?:19|20)[0-9]{2}(?:(?:0?[0-9])|(?:1[012]))(?:(?:0?[0-9])|1[0-9]|2[0-9]|3[01])T(?:0[0-9]|1[0-9]|2[0-3])(?:[0-5][0-9])(?:[0-5][0-9])(?:Z|(?:-(?:0[0-9]:[03]0)))?",
```

Comment:
What are we to gain from breaking up the message into `DATETIME` and `DATETIME2` ?

(as opposed to combining both messages into "DATETIME")

Reply:
Technically I don't think a datetime format can selectively use dashes between the numbers so I broke down the datetime regexes into one regex for datetimes with dashes and one regex for datetimes without dashes, I did the same thing for time as well, there's a way to combine both of them but I think it's actually more simple to have it be this way, or else the regex may become too long or too complicated.

what might be a good idea is to allow the values in the dict to be lists of regexes instead of just one regex, so we can avoid having to make multiple keys, but then allowing users to customize this existing dict becomes complicated (do we append to the list, do we replace the list entirely? Right now the behaviour is that if a user hasa key in their dict that matches a key in the predefined dict, it will replace the value in the predefined dict with theirs).

Reply:
I like turning the values into lists and having less keys.

I feel that combining user provided regexes into the list would be the best idea, and we can put those first, so if it's gonna match it'll match the user defined regex first.